Core/contentblocker.js

```diff
@@ -553,29 +553,47 @@ _utf8_encode : function (string) {
     // Init
     (function() {
         
-        duckduckgoDebugMessaging.log("installing beforeload detection")
-        document.addEventListener("beforeload", function(event) {
-
-            if (event.target.nodeName == "LINK") {
-                type = event.target.rel
-            } else if (event.target.nodeName == "IMG") {
-                type = "image"
-            } else if (event.target.nodeName == "IFRAME") {
-                type = "subdocument"
-            } else {
-                type = event.target.nodeName
-            }
-
-            duckduckgoDebugMessaging.log("checking " + event.url + " (" + type + ")");
-            if (shouldBlock(event.url, type)) {
-                duckduckgoDebugMessaging.log("blocking beforeload")
-                event.preventDefault()
-                event.stopPropagation()
-            } else {
-                duckduckgoDebugMessaging.log("don't block " + event.url);
-                return
-            }
-        }, true)
+        duckduckgoDebugMessaging.log("installing load detection")
+        window.addEventListener("load", function(event) {
+            [].slice.apply(document.scripts).forEach(function(el) {
+                if (shouldBlock(el.src, 'SCRIPT')) {
+                    duckduckgoDebugMessaging.log("blocking load")
```

Comment:
As the load events happen this will continue logging for each load right which is much more than before.

Reply:
Likely. But does not appear to have any detrimental side effects in my testing. There's already a lot of logging going on.